### DISCUSSION TITLE
Welcome to Mixture Discussions!

### DISCUSSION THREAD

**Discussion by alelievr** [alelievr/Mixture#19] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by felixcantet** [comment]
Hello,

I am a French Students in Video Game and 3D ingeneering. For our end of study project, we were trying with two partners to develop a procedural texture generator in Unity, based on NodeGraphProcessor. When Mixture was released, we reoriented our project to a contribution to Mixture because a community project is much more interesting than a personnal project.

We are currenty exploring Mixture and thinking about the object of our contribution. We are very interested in workflow enhancement 
 like a procedural painting module (Quixel Mixer) and Machine Learning based solutions for texture processing. 

Sorry for my bad english 

Thank you for this great project

Félix Cantet

**Reply by alelievr** [reply]
Hello and welcome!

Thanks for your interest for Mixture :) If you want to contribute to Mixture, I can help you find a good design/implementation that would fit in the scope of the project and integrate correctly with other features.

When you talk about Quixel Mixer I guess you mean the layer combining part of it? The material workflow has yet to be determined in Mixture and it indeed needs several nodes that would operate on the material level instead of just textures. I know there is a lot of work in this area to do and it's definitely something that will be needed/useful in the future.

About machine learning, I already thought about it and the problem I have with this is how slow it is (not possible for realtime effects), most of the time requires several libraries to run instead of being a standalone thing and the parameters exposed from a neural network are generally fuzzy and hard to control. I'm not saying that machine learning is not useful, it can definitely have great results for tiling, upscaling, in-painting, or style transfer but it's generally hard to maintain.   
Personally, I prefer this kind of approach: https://www.youtube.com/watch?v=kFLZiak5gyc (from a tech artist in Embark Studio) that is standalone and I believe can achieve real-time performances with a GPU oriented implementation (I don't want to influence you at all, it's just what I think about it 🙂 )

**Reply by felixcantet** [reply]
Thank you for your feedback,

Yes, the reference to Quixel Mixer is the material layer combining part and 3D mask painting. We will for sure talk with you to design this development. We are currently totally busy with multiple schoold project, and we plan to begin the active developement in February.

One important part of our school project is to write a small Siggraph paper (officialy named Poster), and the Machine Learning part is an application for this paper. We plan to use Mixture as a base for this implementation, but maybe it will not fit with the project, in that case we will not publish this part of our work. We are currently working on a Neural Network able to extract multiple masks from an image to identify the different microsurfaces that compose it.

One precision, for all of us in my group, Mixture will be our first Open Source contribution, so we are not very familiar with this developement process

Thank you for your feedback

Have a nice day

**Reply by alelievr** [reply]
Great, don't hesitate to ask me if you have any question :)

looking forward to working with you!
